okta/resource_okta_user.go

```diff
@@ -423,6 +471,49 @@ func resourceUserUpdate(d *schema.ResourceData, m interface{}) error {
 		}
 		d.SetPartial("group_memberships")
 	}
+
+	if passwordChange {
+		oldPassword, newPassword := d.GetChange("password")
+
+		op := &okta.PasswordCredential{
+			Value: oldPassword.(string),
+		}
+		np := &okta.PasswordCredential{
+			Value: newPassword.(string),
+		}
+		npr := &okta.ChangePasswordRequest{
+			OldPassword: op,
+			NewPassword: np,
+		}
+
+		_, _, err := client.User.ChangePassword(d.Id(), *npr, nil)
+		if err != nil {
+			return fmt.Errorf("[ERROR] Error Updating User password in Okta: %v", err)
+		}
+	}
+
+	if recoveryQuestionChange || recoveryAnswerChange {
+		p := &okta.PasswordCredential{
+			Value: d.Get("password").(string),
+		}
+
+		rq := &okta.RecoveryQuestionCredential{
+			Question: d.Get("recovery_question").(string),
+			Answer:   d.Get("recovery_answer").(string),
+		}
+
+		nuc := &okta.UserCredentials{
+			Password:         p,
+			RecoveryQuestion: rq,
+		}
+
+		_, _, err := client.User.ChangeRecoveryQuestion(d.Id(), *nuc)
```

Comment:
@quantumew - The recovery credentials update doesn't seem to work in my test environment. Okta APIs don't throw an error but the credentials are not updated in the user account. (Password is updated, but recovery question/answer is not). Trying to debug the issue. Don't merge this yet.